### PR TITLE
Default store location endpoint for active plugin

### LINKED ISSUE #5335: Update the app to use /terminal/locations/store endpoint of the active plugin (WCPay vs Stripe Extension)
Update the app to use  /terminal/locations/store endpoint of the active plugin (WCPay vs Stripe Extension).

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Merge instructions:

1. Review this PR
2. Make sure https://github.com/woocommerce/woocommerce-android/pull/5509 is merged
2. Merge fluxc PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2224
3. Update fluxc hash in this PR
4. Remove "Do not merge" label
5. Merge this PR

Closes: #5335 

### Description

This PR updates WCAndroid so it's compatible with fluxC after introduction of `/wc_stripe/terminal/locations/store` endpoint - https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2224

### Testing instructions

1. Install the newest version of the Stripe Extension plugin to your store (the version hasn't been released yet, you'll need to build it locally)
2. Make sure WCPay is not active in your store
3. Open the App
4. Open app settings
5. Tap on "In-Person Payments"
6. Tap on "Manage Card Reader"
7. Tap on the "Connect card reader" button
8. Check in the debugger that `/wc_stripe/terminal/locations/store` endpoint is invoked. You can use a flipper or a similar tool for this as well.

P.S. I have only tested this via flipper.

### Images/gif

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/connect/CardReaderConnectViewModel.kt [hunk 1]
```diff
@@ -395,6 +396,23 @@ class CardReaderConnectViewModel @Inject constructor(
         }
     }
 
+    private fun getPaymentPluginType(): PluginType = when (getCardReaderOnboardingCompletedStatus()) {
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/connect/CardReaderConnectViewModel.kt]: 💡 We might want to consider introducing a new getter in the AppPrefs and return `PluginType?`. It feels a bit off that mapping from `PluginType` to `CardReaderOnboardingCompletedStatus` lives in the AppPrefs but the opposite mapping lives in the client of AppPrefs. Wdyt?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/connect/CardReaderConnectViewModel.kt [hunk 2]
```diff
@@ -395,6 +396,23 @@ class CardReaderConnectViewModel @Inject constructor(
...
+        AppPrefs.CardReaderOnboardingCompletedStatus.CARD_READER_ONBOARDING_COMPLETED_WITH_WCPAY ->
+            PluginType.WOOCOMMERCE_PAYMENTS
+        AppPrefs.CardReaderOnboardingCompletedStatus.CARD_READER_ONBOARDING_COMPLETED_WITH_STRIPE_EXTENSION ->
+            PluginType.STRIPE_TERMINAL_GATEWAY
+        AppPrefs.CardReaderOnboardingCompletedStatus.CARD_READER_ONBOARDING_NOT_COMPLETED ->
+            throw IllegalStateException("Onboarding not completed. Plugin Type is null")
+    }
+
+    private fun getCardReaderOnboardingCompletedStatus(): AppPrefs.CardReaderOnboardingCompletedStatus {
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/connect/CardReaderConnectViewModel.kt]: 💡 We might want to consider moving this logic into the LocationRepository - I feel like the client of the repository shouldn't need to know anything about a currently active plugin. Wdyt?

**Review by malinajirka — APPROVED** [review]
Thanks for these changes @AnirudhBhat! I've left a couple comments you might want to consider - they are not vital, so feel free to merge this PR as is.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/connect/CardReaderConnectViewModel.kt (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/connect/CardReaderConnectViewModel.kt]: Good point! Moved it here https://github.com/woocommerce/woocommerce-android/pull/5500/commits/bd4e19b780e9ef201a614821c7c01f581dda2e8b

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/connect/CardReaderConnectViewModel.kt (hunk 2 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/connect/CardReaderConnectViewModel.kt]: Agree! Fixed here https://github.com/woocommerce/woocommerce-android/pull/5500/commits/bd4e19b780e9ef201a614821c7c01f581dda2e8b

**Comment by AnirudhBhat** [comment]
@malinajirka Thanks for the review! I've updated the base branch for this PR to another PR which has a fix for onboarding completed status issue which we discussed on slack. Let's merge this once the base PR is merged.

**Review by malinajirka — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**